app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -294,28 +327,67 @@ class StateFragmentPresenter @Inject constructor(
     return fragment.childFragmentManager.findFragmentByTag(TAG_AUDIO_FRAGMENT)
   }
 
-  private fun showHideAudioFragment(isVisible: Boolean) {
-    getStateViewModel().setAudioBarVisibility(isVisible)
+  private fun setAudioFragmentVisible(isVisible: Boolean) {
     if (isVisible) {
-      if (getAudioFragment() == null) {
-        val audioFragment = AudioFragment.newInstance(explorationId, currentStateName)
-        fragment.childFragmentManager.beginTransaction().add(
-          R.id.audio_fragment_placeholder, audioFragment,
-          TAG_AUDIO_FRAGMENT
-        ).commitNow()
-      }
-
-      val currentYOffset = binding.stateRecyclerView.computeVerticalScrollOffset()
-      if (currentYOffset == 0) {
-        binding.stateRecyclerView.smoothScrollToPosition(0)
-      }
+      showAudioFragment()
     } else {
-      if (getAudioFragment() != null) {
-        fragment.childFragmentManager.beginTransaction().remove(getAudioFragment()!!).commitNow()
+      hideAudioFragment()
+    }
+  }
+
+  private fun showAudioFragment() {
+    getStateViewModel().setAudioBarVisibility(true)
+    autoPlayAudio = true
+    (fragment.requireActivity() as AudioButtonListener).showAudioStreamingOn()
+    val currentYOffset = binding.stateRecyclerView.computeVerticalScrollOffset()
+    if (currentYOffset == 0) {
+      binding.stateRecyclerView.smoothScrollToPosition(0)
+    }
+    getAudioFragment()?.let {
+      (it as AudioFragmentInterface).setVoiceoverMappings(explorationId, currentStateName, feedbackId)
```

Comment:
In general, I'm having a lot of difficulty following the logic here. We are setting voiceover mappings in a bunch of places--can that be simplified? E.g. it would be ideal if we just set all voiceover mappings for the state once, then tell the audio manager to play certain content IDs at certain times. As it stands, I'm really concerned that changing some of this logic around will have subtle breakages that are hard to track down. We should try to make the implementation robust by isolating important arrangement code to just one place, and lean on special cases to properly change state (e.g. playing feedback audio should only involve operating with something related to feedback, like a content ID, rather than setting a bunch of state to make it work).

Happy to discuss this further if needed.

Reply:
Changed API of AudioUiManager to have setStateAndExplorationId and loadAudio. This simplified the logic inside StateFragment so that it only has to update state for the AudioFragment and load audio when it needs to based on content id.